### DISCUSSION TITLE
Does the mfem support the sets of materials?

### DISCUSSION THREAD

**Discussion by SkivHisink** [mfem/mfem#2457] [Q&A] [answered]
Does the mfem support the sets of materials for the model when solving the problem of linear elasticity of a finite element assembly?
I have 3D objects with properties Pore ​​pressure, Poisson's ratio, Azimuth of maximum voltage, Density, Maximum and minimum voltage factor and Young's modulus. I need to find Stress and strain tensor in elements by solving elastic linear problem by the finite element method.

**Comment by tzanio** [comment, marked as answer]
We do support materials with different `Poisson's ratio` and `Young's modulus` as shown in Example 2. I am not sure what the other properties you have listed mean though.

**Reply by SkivHisink** [reply]
Actually, I have material for each cell of my object and it number of cells can be more than int32 max value. As I understand, your library can work only with objects that has no more than int32 max value materials, doesn't it?

**Reply by tzanio** [reply]
Are you saying you have more than 2B different materials? If that's the case, then yes, the current material attributes are restricted to size int32.

However, we definitely support more than 2B elements and it could be possible to still express the materials you have, not with the mesh attributes, but a specially designed `Coefficient`.

**Reply by SkivHisink** [reply]
Where can I find example with ``Coefficient``?

**Reply by tzanio** [reply]
It is used e.g. in Example 2, https://github.com/mfem/mfem/blob/master/examples/ex2.cpp#L144-L153
